File: app/src/main/java/com/jwu5/giphyapp/giphyfavorites/FavoritesPresenter.java
package com.jwu5.giphyapp.giphyfavorites;

import android.content.Context;

import com.jwu5.giphyapp.dataSingleton.SavedFavorites;
import com.jwu5.giphyapp.network.model.GiphyModel;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.Set;

/**
 * Created by Jiawei on 8/29/2017.
 */
public class FavoritesPresenter {
    private final FavoritesView mView;
    private Context mContext;
    private SavedFavorites mFavorites;

    public FavoritesPresenter(FavoritesView view, Context context) {
        mView = view;
        mContext = context;
        mFavorites = SavedFavorites.getInstance();
    }

    public LinkedHashMap<String, GiphyModel> getFavorites() {
        return mFavorites.getFavorites();
    }

    public void updateUI(int position) {
        LinkedHashMap<String, GiphyModel> tempFavorites = mFavorites.getFavorites();
        ArrayList<GiphyModel> tempList = new ArrayList<>();
        Set<String> keys = tempFavorites.keySet();

        for (String k : keys) {
            tempList.add(tempFavorites.get(k));
        }

        mView.updateAdapter(tempList, position);
    }
}

File: app/src/main/java/com/jwu5/giphyapp/giphyfavorites/FavoritesPresenter.kt
package com.jwu5.giphyapp.giphyfavorites

import android.content.Context

import com.jwu5.giphyapp.dataSingleton.SavedFavorites
import com.jwu5.giphyapp.network.model.GiphyModel

import java.util.ArrayList
import java.util.LinkedHashMap

/**
 * Created by Jiawei on 8/29/2017.
 */
class FavoritesPresenter(private val mView: FavoritesView, private val mContext: Context) {
    private val mFavorites: SavedFavorites

    init {
        mFavorites = SavedFavorites.instance
    }

    val favorites: LinkedHashMap<String, GiphyModel>
        get() = mFavorites.favorites!!

    fun updateUI(position: Int) {
        val tempFavorites = mFavorites.favorites
        val tempList = ArrayList<GiphyModel>()
        val keys = tempFavorites!!.keys

        for (k in keys) {
            tempList.add(tempFavorites[k]!!)
        }

        mView.updateAdapter(tempList, position)
    }
}
